src/serializers/fields.rs

```diff
@@ -156,56 +156,35 @@ impl GeneralFieldsSerializer {
         let output_dict = PyDict::new(py);
         let mut used_req_fields: usize = 0;
 
-        // NOTE! we maintain the order of the input dict assuming that's right
-        for result in main_iter {
-            let (key, value) = result?;
-            let key_str = key_str(&key)?;
-            let op_field = self.fields.get(key_str);
-            if extra.exclude_none && value.is_none() {
-                if let Some(field) = op_field {
-                    if field.required {
+        if !extra.sort_keys {
```

Comment:
Can we swap these branches so we have the true case first followed by the false case? Same for the other instance of `if !extra.sort_keys`